lab-justin/lib/parse-body.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+module.exports = function(req){
+  return new Promise(function(resolve, reject){
+    if(/(POST|PUT|DELETE)/.test(req.method)){
+      req.body = '';
+      req.on('data', function(data){
+        //append to the strinf +=
+        req.body += data.toString();
+      });
+
+      req.on('end', function(){
+        try {
+          req.body = JSON.parse(req.body);
+          resolve();
+        } catch(err){
+          reject(err);
+        }
+      });
+      return;
```

Comment:
Stylistically, it might make sense to do return resolve() and return reject(err), that way it's not necessary to look super closely at the nesting of the curly braces to figure out exactly when this happens.